Inline asm: `*const [type error]` does not implement the Copy trait

### Code

```Rust
fn main() {
	let peb: *const PEB;
	unsafe { std::arch::asm!("mov {0}, fs:[0x30]", out(reg) peb); }
}
```


### Current output

```Shell
error[E0412]: cannot find type `PEB` in this scope
 --> src/main.rs:2:18
  |
2 |     let peb: *const PEB;
  |                     ^^^ not found in this scope

error: arguments for inline assembly must be copyable
 --> src/main.rs:3:49
  |
3 |     std::arch::asm!("mov {0}, fs:[0x30]", out(reg) peb);
  |                                                    ^^^
  |
  = note: `*const [type error]` does not implement the Copy trait

For more information about this error, try `rustc --explain E0412`.
```


### Desired output

```Shell
Probably just omit the second error; no `Copy` error is given when moving an unknown type normally.
```


### Rationale and extra context

_No response_

### Other cases

_No response_

### Anything else?

Happens on `1.73.0-nightly (2023-07-16 0e8e857b11f60a785aea)`, as well as several earlier versions.

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"RickleAndMortimer"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_hir_analysis/src/check/intrinsicck.rs
+++ b/compiler/rustc_hir_analysis/src/check/intrinsicck.rs
@@ -68,7 +68,7 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
         let asm_ty = match *ty.kind() {
             // `!` is allowed for input but not for output (issue #87802)
             ty::Never if is_input => return None,
-            ty::Error(_) => return None,
+            _ if ty.references_error() => return None,
             ty::Int(IntTy::I8) | ty::Uint(UintTy::U8) => Some(InlineAsmType::I8),
             ty::Int(IntTy::I16) | ty::Uint(UintTy::U16) => Some(InlineAsmType::I16),
             ty::Int(IntTy::I32) | ty::Uint(UintTy::U32) => Some(InlineAsmType::I32),

--- a/tests/ui/asm/issue-113788.rs
+++ b/tests/ui/asm/issue-113788.rs
@@ -1,0 +1,7 @@
+// test that "error: arguments for inline assembly must be copyable" doesn't show up in this code
+// needs-asm-support
+// only-x86_64
+fn main() {
+    let peb: *const PEB; //~ ERROR cannot find type `PEB` in this scope [E0412]
+    unsafe { std::arch::asm!("mov {0}, fs:[0x30]", out(reg) peb); }
+}

--- a/tests/ui/asm/issue-113788.stderr
+++ b/tests/ui/asm/issue-113788.stderr
@@ -1,0 +1,9 @@
+error[E0412]: cannot find type `PEB` in this scope
+  --> $DIR/issue-113788.rs:5:21
+   |
+LL |     let peb: *const PEB;
+   |                     ^^^ not found in this scope
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0412`.

